src/travel-details-map-screen/TravelDetailsMapScreenComponent.tsx

```diff
@@ -159,71 +175,108 @@ const styles = StyleSheet.create({
   mode?: AnyMode;
   subMode?: AnySubMode;
   setShouldTrack: React.Dispatch<React.SetStateAction<boolean>>;
+  subscriptionState: SubscriptionState;
+  zoomLevel: number;
 };
 
 const LiveVehicle = ({
-  coordinates,
+  vehicle,
   setShouldTrack,
   mode,
   subMode,
+  subscriptionState,
+  zoomLevel,
 }: VehicleIconProps) => {
+  const {theme} = useTheme();
+
+  const isError =
+    subscriptionState === 'CLOSING' || subscriptionState === 'CLOSED';
+  const isLoading =
+    subscriptionState === 'CONNECTING' || subscriptionState === 'NOT_STARTED';
+  const [isStale, setIsStale] = useState(false);
+
+  useInterval(
+    () => {
+      const secondsSinceUpdate = secondsBetween(
+        vehicle.lastUpdated,
+        new Date(),
+      );
+      setIsStale(STALE_DATA_THRESHOLD_IN_SECONDS < secondsSinceUpdate);
+    },
+    1000,
+    [vehicle.lastUpdated],
+    false,
+    true,
+  );
+
   const circleColor = useTransportationColor(mode, subMode);
-  const iconName = vehicleIconName(mode);
+  const textColor = useTransportationColor(mode, subMode, 'text');
+  const svg = getTransportModeSvg(mode, subMode);
+
+  const circleStyle = ((): CircleLayerStyleProps => {
+    if (isError)
+      return {
+        circleColor:
+          theme.interactive.interactive_destructive.disabled.background,
+        circleRadius: 20,
+        circleStrokeColor:
+          theme.interactive.interactive_destructive.default.background,
+        circleStrokeWidth: 2,
+      };
+    if (isStale)
+      return {
+        circleColor: theme.interactive.interactive_1.disabled.background,
+        circleRadius: 22,
+        circleStrokeWidth: 0,
+      };
+    return {
+      circleColor,
+      circleRadius: 22,
+      circleStrokeWidth: 0,
+    };
+  })();
 
-  if (!coordinates) return null;
+  const LiveVehicleIcon = (): JSX.Element => {
+    if (isLoading) return <ActivityIndicator color={textColor} />;
+    if (isError)
+      return (
+        <ThemeIcon
+          svg={svg}
+          fill={theme.interactive.interactive_destructive.default.background}
+        />
+      );
+    if (isStale)
+      return (
+        <ThemeIcon
+          svg={svg}
+          fill={theme.interactive.interactive_1.disabled.text}
+        />
+      );
+    return <ThemeIcon svg={svg} fill={textColor} />;
+  };
+
+  if (!vehicle.location || zoomLevel < FOLLOW_MIN_ZOOM_LEVEL) return null;
   return (
-    <MapboxGL.ShapeSource
-      id="liveVehicle"
-      shape={pointOf(coordinates)}
-      cluster
-      onPress={() => {
-        setShouldTrack(true);
-      }}
-    >
-      <MapboxGL.CircleLayer
-        id="liveVehicleCircle"
-        minZoomLevel={8}
-        style={{
-          circleColor,
-          circleRadius: 22,
-        }}
-      />
-      <MapboxGL.SymbolLayer
+    <>
+      <MapboxGL.ShapeSource id="liveVehicle" shape={pointOf(vehicle.location)}>
+        <MapboxGL.CircleLayer id="liveVehicleCircle" style={circleStyle} />
+      </MapboxGL.ShapeSource>
```

Comment:
You're probably right, but this does complicate things a bit, and it might need to change when we add the direction arrow. I think I'll leave it as is for now, then revisit it when we add the arrow.

Reply:
Aight. I though it was just adding a background and border radius to the TouchableOpacity, but if it not that easy it is ok to wait.